transport/src/main/java/io/netty/channel/nio/AbstractNioChannel.java

```diff
@@ -353,6 +353,7 @@ protected void doRegister() throws Exception {
     @Override
     protected void doDeregister() throws Exception {
         eventLoop().cancel(selectionKey());
+        selectionKey=null;
```

Comment:
@normanmaurer  so is it really one bug? I use netty 4.0.33 final in my project and found can't gc it. I prepare to test more cases and what's more, i want to know if it is one bug? if so i will improve my code to fix it.